OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalNotificationManager.java

```diff
@@ -102,7 +112,23 @@ static void assignGrouplessNotifications(Context context, ArrayList<StatusBarNot
             Notification notif = grouplessNotifBuilder
                     .setGroup(GROUPLESS_SUMMARY_KEY)
                     .build();
-            NotificationManagerCompat.from(context).notify(grouplessNotif.getId(), notif);
+
+            notifyNotificationUpdate(context, grouplessNotif.getId(), notif);
+        }
+    }
+
+    /**
+     * Separated notify() calls with NotificationManagerCompat to help various places consolidate
+     * to a single method call, which will help catching any fatal exceptions and prevent crashes
+     */
+    static void notifyNotificationUpdate(Context context, int notifId, Notification notification) {
+        try {
+            NotificationManagerCompat.from(context).notify(notifId, notification);
+        } catch(Throwable e) {
```

Comment:
1. Instead of catching everything I think we should just catch `android.os.DeadSystemException` here as some of the reports we have gotten are throwing.
2. Instead of silently eating this we should also log it as a warning that the notification could be be shown due to the exception too.